src/services/dashboards/dashboard-customize/modules/dashboard-widget-input-form/schema-helpers.ts

```diff
@@ -100,8 +103,15 @@ export const getRefinedWidgetOptionsSchema = (
     Object.entries(schema.properties).forEach(([propertyName, propertySchema]) => {
         // set properties declared in defaultSchemaProperties only
         if (!defaultSchemaProperties.includes(propertyName)) return;
+        const referenceType = propertyName.replace('filters.', '');
         const isInherit = !!inheritOptions[propertyName]?.enabled;
         refinedJsonSchema.properties[propertyName] = getWidgetOptionSchema(propertyName, propertySchema, variablesSchema, referenceStoreState, isInherit);
+        if (projectId && referenceType === REFERENCE_TYPE_INFO.project.type) {
+            refinedJsonSchema.properties[propertyName] = {
+                ...refinedJsonSchema.properties[propertyName],
+                disabled: true,
+            };
+        }
```

Comment:
suggestion: I think the logics for project dashboard case must also be applied to inherit case.
What about to apply this logic to `getWidgetOptionSchema`?